src/util/dataAccess.cpp

```diff
@@ -324,6 +324,43 @@ bool positionAndExtentInData(const DataArray &data, const NDSize &position, cons
 }
 
 
+DataView dataSlice(const DataArray &array, const std::vector<double> &start, const std::vector<double> &end,
+                   const std::vector<std::string> &units) {
+    if (array == nix::none) {
+        throw UninitializedEntity();
+    }
+    if (start.size() == 0 || start.size() != end.size()) {
+        throw OutOfBounds("Number of start entries does not match number of end entries.", 0);
+    }
+    if (start.size() != array.dimensionCount()) {
+        throw OutOfBounds("Number of start/end entries does not match dimensionality of the data.", 0);
```

Comment:
We could pass `start.size()` as the "where" parameter (currently `0`). But maybe that is confusing.

Reply:
Actually, I think `OutOfBounds` is probably not 100% the right exception, but I don't think that is crucial.